Rakefile

```diff
@@ -1,24 +1,19 @@
 require 'rake'
 require 'rake/testtask'
 
+require_relative 'lib/tasks/exercise'
+require_relative 'lib/tasks/exercise_test_tasks'
 require_relative 'lib/tasks/exercise_tests_runner'
-require_relative 'lib/tasks/knows_what_exercises_are_called_and_where_they_live_in_the_directory_structure'
+require_relative 'lib/tasks/rake_arguments'
 
-Exercise.all.each do |exercise|
-  task exercise do
-    ExerciseTestsRunner.new(
-      exercise: exercise,
-      test_options: ARGV.drop_while { |e| e != '--' }.drop(1).join(' ')
-    ).run
-  end
-end
+ExerciseTestTasks.new exercises: Exercise.all,
+                      options: RakeArguments.task_options
 
 task default: 'test'
 
 desc 'Run individual exercises or run all development and exercise tests'
 task :test do
-  exercises = ARGV.take_while { |e| e != '--' }.drop(1)
-  next if exercises.any?
+  next if RakeArguments.more_tasks?
```

Comment:
I still prefer wrapping the steps below in a conditional, to using `next`, but that may just be an unimportant stylistic preference.

I like blocks to have have a single exit point at the end, unless there is a **really** good reason not to do that.